sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -225,15 +245,25 @@ public override string ToString()
         /// <returns>A stream representing the data.</returns>
         public Stream ToStream()
         {
+            if (_originalStream != null)
```

Comment:
Should we always return a new `MemoryStream` here? If the user already called `ToBytes`, their stream's position would not be 0.